torch_frame/utils/skorch.py

```diff
@@ -0,0 +1,313 @@
+from __future__ import annotations
+
+import importlib
+import warnings
+from typing import Any
+
+import numpy as np
+import skorch.utils
+import torch
+from numpy.typing import ArrayLike, NDArray
+from pandas import DataFrame
+from sklearn.model_selection import train_test_split
+from skorch import NeuralNet
+from torch import Tensor
+
+import torch_frame
+from torch_frame import nn
+from torch_frame.config import (
+    ImageEmbedderConfig,
+    TextEmbedderConfig,
+    TextTokenizerConfig,
+)
+from torch_frame.data.dataset import Dataset
+from torch_frame.data.loader import DataLoader
+from torch_frame.data.tensor_frame import TensorFrame
+from torch_frame.typing import IndexSelectType
+from torch_frame.utils import infer_df_stype
+
+
+# TODO: make it more safe
+def _patch_skorch_support_tenforframe() -> None:
+    old_to_tensor = skorch.utils.to_tensor
+
+    def to_tensor(X, device, accept_sparse=False):
+        if isinstance(X, TensorFrame):
+            return X
+        return old_to_tensor(X, device, accept_sparse)
+
+    skorch.utils.to_tensor = to_tensor
+
+    importlib.reload(skorch.net)
+
+
+_patch_skorch_support_tenforframe()
+
+
+class NeuralNetPytorchFrameDataLoader(DataLoader):
+    def __init__(self, dataset: Dataset | TensorFrame, *args,
+                 device: torch.device, **kwargs):
+        super().__init__(dataset, *args, **kwargs)
+        self.device = device
+
+    def collate_fn(  # type: ignore
+            self, index: IndexSelectType) -> tuple[TensorFrame, Tensor | None]:
+        index = torch.tensor(index)
+        res = super().collate_fn(index).to(self.device)
+        return res, res.y
+
+
+class NeuralNetPytorchFrame(NeuralNet):
+    def __init__(
+        self,
+        # NeuralNet parameters
+        module,
```

Comment:
Also add the typing information here?

Reply:
Since skorch does not support typing, doing this is tedious and not recommended.